lib/vanagon/platform/dsl.rb

```diff
@@ -161,6 +161,22 @@ def servicetype(type)
         @platform.servicetype = type
       end
 
+      # Set the list of possible host to perform a build on (when not using
+      # pooler or CLI flags)
+      #
+      # @param type [Array] the names of the hosts (must be resolvable)
+      def build_hosts(*args)
+        raise ArgumentError if args.length < 1
+        if args[0].is_a?(Array)
```

Comment:
This can be simplified to:
`@platform.build_hosts = Array(args).flatten`

That way is args is an array, it will get re-arrayed and then de-arrayed, and if it is just a string it will be an array of 1 element.